### PR TITLE
Alternate way of adding metadata to cloned tasks

### PR DESCRIPTION
**Pull request by romain-intel**
This adds it on the client side

### CODE REVIEW

**Comment by romain-intel** [comment]
This replaces #1002

File: metaflow/client/core.py [hunk 1]
```diff
@@ -1280,23 +1323,28 @@ def environment_info(self):
         if not env_type:
             return None
         env = [m for m in ENVIRONMENTS + [MetaflowEnvironment] if m.TYPE == env_type][0]
-        return env.get_client_info(self.path_components[0], self.metadata_dict)
+        meta_dict = self.metadata_dict
+        return env.get_client_info(self.path_components[0], meta_dict)
```
Comment (savingoyal) [inline metaflow/client/core.py:1327]: How is this change materially different from what existed before?

Comment (romain-intel) [reply metaflow/client/core.py:1327]: Right, it isn't. Sorry, a bit too trigger happy in my changes. I'll revert it back.

**Review by savingoyal — APPROVED** [review]
LGTM! Let's merge this in as soon as OB/NFLX tests pass.

**Pull request merged**